src/main/java/nextstep/ladder/entity/PositionMoveMatch.java

```diff
@@ -0,0 +1,29 @@
+package nextstep.ladder.entity;
+
+import java.util.Arrays;
+import java.util.function.Function;
+
+public enum PositionMoveMatch {
+
+	NOT_MOVE((root, rootIndex, position) -> !root.getRoot(), Position::new),
+	MOVE_RIGHT((root, rootIndex, position) -> root.getRoot() && rootIndex.equals(position), (position) -> new Position(position + 1)),
+	MOVE_LEFT((root, rootIndex, position) -> root.getRoot() && rootIndex + 1 == position, (position) -> new Position(position - 1)),
+	DEFAULT((root, rootIndex, position) -> true, Position::new);
+
+	private final TriFunction<Root, Integer, Integer, Boolean> condition;
```

Comment:
개인적으로는 3번째 파라미터 타입을 Integer가 아닌 Position으로 관리하셨으면 더 좋았을 것 같아요!
왜냐하면 Integer라는 값은 도메인간 어떤 의미를 가지는지 알 수 없는 단순 스칼라 값이기 때문에
팀원이나 다른 사람이 해당 객체를 사용할 때 연속되는 파라미터들인 Integer의 값을 구분짓고 
해당 파라미터는 어떤 값이며 왜 쓰고있는지 정보를 전할 수 있도록 의미있는 값으로 다룰 수 있었으면 더 좋았을 것 같아요!
또한 클린 코드 과정이 객체지향 패러다임을 준수하는 방향에 맞게 값이 아닌 객체로 다룬다면
Position 내부의 값을 외부에 노출을 방지할 수 있게되어 캡슐화도 지켜질 수 있다고 생각합니다 😄 

```suggestion
	NOT_MOVE((root, rootIndex, position) -> !root.getRoot(), Position::new),
	MOVE_RIGHT((root, rootIndex, position) -> root.getRoot() && position.canMoveRight(rootIndex), Position::moveRight),
	MOVE_LEFT((root, rootIndex, position) -> root.getRoot() && position.canMoveLeft(rootIndex), Position::moveLeft),
	DEFAULT((root, rootIndex, position) -> true, Position::new);

	private final TriFunction<Root, Integer, Position, Boolean> condition;
```

위 코드는 예시이며 해당 의견들은 정답이 아닌 개인적인 의견이기에 참고만 해주세요 ㅎㅎ
연빈님의 의견은 어떠하신지 궁금하네요..! 👍

Reply:
맞습니다 저도 동일하게 생각합니다.

값에의한 control이 아닌, 객체에게 메세지를 던져라. 라는 뉘앙스로 작업을 햇으면 위와 같은 코드가 나오지않았을텐데요

제가 놓쳤던 부분인것같습니다.. ㅎ

Position이동은 위와같이 진행을 하면될것 같구요
Parameter는 guard하게 Position을 주는것이 더 알맞는것 같습니다 
피드백 감사합니다 😄